test/integration_test.py

```diff
@@ -737,15 +790,30 @@ def _validate_fastq_content(self, content: ReadableFileObject):
         self.assertTrue(lines[0].startswith(b'@'))
         self.assertTrue(lines[2].startswith(b'+'))
 
-    def _prepare_notifications(self, catalog: CatalogName) -> Dict[BundleFQID, JSON]:
-        bundle_fqids = self._list_bundles(catalog,
-                                          max_bundles=self.max_bundles,
-                                          check_all=True)
-        log.info('Preparing notifications for catalog %r.', catalog)
-        return {
-            bundle_fqid: self.azul_client.synthesize_notification(bundle_fqid)
-            for bundle_fqid in bundle_fqids
-        }
+    def _prepare_notifications(self, catalog: CatalogName) -> Tuple[JSONs,
+                                                                    Set[SourcedBundleFQID]]:
+        bundle_fqids = set()
+        notifications = []
+
+        def update(source: SourceRef,
+                   prefix: str,
+                   partition_bundle_fqids: Iterable[SourcedBundleFQID]):
+            bundle_fqids.update(partition_bundle_fqids)
+            notifications.append({
+                'action': 'reindex',
+                'catalog': catalog,
+                'source': source.to_json(),
+                'prefix': prefix
+            })
+
+        list(starmap(update, self._list_managed_access_bundles(catalog)))
+        num_bundles = self.min_bundles - len(bundle_fqids)
+        log.info('Selected %d bundles to satisfy managed access coverage; '
+                 'selecting at least %d more', len(bundle_fqids), num_bundles)
+        list(starmap(update, self._list_bundles(catalog,
```

Comment:
> If an MA bundle is added to from line 809, then it's selected again in this _list_bundles method can that be problematic?

The prepared notifications should satisfy the requirements to index some managed bundles before any duplicates are added. Adding duplicates shouldn't have any effect on the indexing results. So I can't imagine what problems this would cause.

Reply:
> There is a comment that says

    To index them with remote_reindex and avoid collateral bundles, we use as specific a
    a prefix as possible.

but if its added a second time there might be those collateral bundles added in.

The duplicates added at the end of the `_prepare_notifications` methods use single-bundle notifications (`'action': 'add'`) instead of prefix-based partition notifications (`'action': 'reindex'`) so the concept of "collateral bundles" does not apply to these intentional duplicates.

Reply:
> On a side note I don't think we have MA projects yet, when we do get production MA projects they will probably be presented as one source per project

We have MA projects on `dev`/`sandbox`, which already use one source per project. So that is exactly what this code was designed to handle.

> I think that could change how picky _list_managed_access_bundles can be about its bundles.

Don't understand what you mean.